test_scripts/Polices/External_UCS/016_ATF_N_TC_PTU_with_external_consent_status_groups_struct.lua

```diff
@@ -0,0 +1,132 @@
+---------------------------------------------------------------------------------------------
+-- Requirement summary:
+-- [Policies] External UCS: PTU with "external_consent_status_groups" struct
+--
+-- Description:
+-- In case:
+-- SDL receives PolicyTableUpdate with "external_consent_status_groups:
+-- [<functional_grouping>: <Boolean>]” -> of "device_data" -> "<device identifier>"
+-- -> "user_consent_records" -> "<app id>" section
+-- SDL must:
+-- a. consider this PTU as invalid
+-- b. do not merge this invalid PTU to LocalPT.
+--
+-- Steps:
+-- 1. Register app
+-- 2. Activate app
+-- 3. Perform PTU (make sure 'external_consent_status_groups' exists in PTU file)
+-- 4. Verify status of update
+-- 5. Verify LocalPT (using data in snapshot)
+--
+-- Expected result:
+-- a) PTU fails with UPDATE_NEEDED status
+-- b) 'external_consent_status_groups' section doesn't exist in LocalPT
+--
+-- Note: Script is designed for EXTERNAL_PROPRIETARY flow
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local testCasesForExternalUCS = require('user_modules/shared_testcases/testCasesForExternalUCS')
+
+--[[ Local variables ]]
+local appId = config.application1.registerAppInterfaceParams.appID
+local grpId = "Location-1"
```

Comment:
@istoimenova, Please check b1d1866
Regarding scripts 010 and 015 - these ones no need to be updated since grpId = "Location-1" is used as a group (not a name of _user_consent_prompt_ like in 016)